### DISCUSSION TITLE
messages in seperate json files

### DISCUSSION THREAD

**Discussion by codeflorist** [nuxt-modules/i18n#1667] [Q&A] [answered]
i do not want to put my messages inside `nuxt.config.js`, but inside locale-specific files inside a `locales` directory. hot do i configuire the nuxt3 module to do this?

**Comment by ineshbose** [comment, marked as answer]
Hi @codeflorist,

You can use v8 for Nuxt 3 that allows you to put translations in separate `json`, `json5` and `yaml` files through lazy-load!
https://v8.i18n.nuxtjs.org/guide/lazy-load-translations

**Reply by codeflorist** [reply]
thanks a lot for the info! `langDir` is working nicely.

`lazy: true` however has following problem (tested in `dev` mode):
- if page is loaded in the default language, switching from default language to another throws `[intlify] Not found ... key in ... locale messages` warnings and does not display translated texts.
- re-loading the page in a non-default language displays the correctly translated texts and switching back and forth works correctly
